frontend/src/pages/_middleware.ts

```diff
@@ -3,15 +3,20 @@ import { NextResponse } from "next/server";
 
 import { DEFAULT_LOCALE, VALID_LOCALES } from "~/locale/languages";
 
+const PUBLIC_FILE = /\.(.*)$/;
+
 export function middleware(req: NextRequest, _ev: NextFetchEvent) {
-  const { pathname } = req.nextUrl;
+  const shouldHandleLocale =
+    !PUBLIC_FILE.test(req.nextUrl.pathname) &&
+    !req.nextUrl.pathname.includes("/api/") &&
+    req.nextUrl.locale === "default";
   const locale = getLocale(req.cookies.pyconLocale);
 
-  if (pathname === "/") {
-    return NextResponse.redirect(`/${locale}`);
-  }
-
-  return NextResponse.next();
+  return shouldHandleLocale
+    ? NextResponse.redirect(
+        `/${locale}${req.nextUrl.pathname.replace("/default", "")}`,
```

Comment:
What is the purpose of this? We really need `default` as a default locale?

Reply:
think this should be the workaround to make Next.js include the `/en/` prefix in the default locale
https://nextjs.org/docs/advanced-features/i18n-routing#prefixing-the-default-locale

Reply:
The reason there is the pathname.replace is because of an issue with how Nextjs returns .href: https://github.com/vercel/next.js/issues/31533